src/test/java/com/fasterxml/jackson/databind/deser/dos/DeepJsonNodeSerTest.java

```diff
@@ -33,7 +33,6 @@ public void testVeryDeepNodeSer() throws Exception
         jsonString.append("}");
 
         JsonNode jsonNode = MAPPER.readTree(jsonString.toString());
-        String json = MAPPER.writeValueAsString(jsonNode);
-        assertNotNull(json);
+        assertNotNull(jsonNode);
```

Comment:
the Stackoverflow that happens after the change that @cowtowncoder made to this test a few days is actually not related to StreamReadConstraints.DEFAULT_MAX_DEPTH - the failure is a different recursion bug -- it is the reason I started work on a separate PR - #4021

With #4021, I have fixed one possible set of stackoverflows but I have found more that even #4021 doesn't yet fix. I have fixes in a private branch for another set of stackoverflows. JsonNode serialization is something that can fail badly if users are careless with the sort of deeply nested or even recursive structures that they create. In the end of the day, Jackson could provide more guard rails but we may not want to overdo it as it could affect performance in a major way. And the new code is more complicated and as a result, possibly more error prone.

Reply:
I think our writer-side limits should work as the ultimate guard. I was under assumption that something in stack size limits might have lower maximum that'd work (since it depends both on frames per output level and stack size settings of runtime environment, presumably with less space for CI).
But maybe that was incorrect assumption. Will check out this PR and 4021 .